airflow/models/dag.py

```diff
@@ -778,16 +778,26 @@ def set_dependency(self, upstream_task_id, downstream_task_id):
     def get_task_instances(
             self, start_date=None, end_date=None, state=None, session=None):
         if not start_date:
+            # TODO why 30?
             start_date = (timezone.utcnow() - timedelta(30)).date()
             start_date = timezone.make_aware(
                 datetime.combine(start_date, datetime.min.time()))
-        end_date = end_date or timezone.utcnow()
-        tis = session.query(TaskInstance).filter(
-            TaskInstance.dag_id == self.dag_id,
-            TaskInstance.execution_date >= start_date,
-            TaskInstance.execution_date <= end_date,
-            TaskInstance.task_id.in_([t.task_id for t in self.tasks]),
-        )
+
```

Comment:
This change to `get_task_instances()` seems to be newly included from the previously closed PR? This is going to change the TI returned to include those considered in the "future".  Why is this change necessary?

If it is indeed needed, you should probably consider doing the same as what you did in `scheduler_job.py`, i.e. only include "future" tasks if `RUN_FUTURE_EXEC_DATES` is True **and** `dag.schedule_interval` is None

Reply:
Okay this does look necessary. And it's easy to fix. How about adding this property to `class DAG`?

```python
class DAG:
    ...
    @property
    def allow_future_exec_dates(self):
        return conf.getboolean('scheduler', 'RUN_FUTURE_EXEC_DATES', fallback=False) and self.schedule_interval is None
```


This line can become:
```python
        filters = [TaskInstance.dag_id == self.dag_id,
                   TaskInstance.execution_date >= start_date,
                   TaskInstance.task_id.in_([t.task_id for t in self.tasks])]

        if end_date or not self.allow_future_exec_dates:
            end_date = end_date or timezone.utcnow()
            filters.append(TaskInstance.execution_date <= end_date)

        tis = session.query(TaskInstance).filter(*filters)
```

And you may want to use `dag.allow_future_exec_dates` in the two other places that you checked the same conditions too, i.e. scheduler_job.py and runnable_exec_date_dep.py

Reply:
updated